code/datums/components/embedded.dm

```diff
@@ -0,0 +1,315 @@
+
+		- Turf embedding is much simpler. All we do here is draw an overlay of the item's inhand on the turf, hide the item, and create an HTML link in the turf's inspect
+		that allows you to rip the item out. There's nothing dynamic about this, so far less checks.
+
+
+	In addition, there are 2 cases of embedding: embedding, and sticking
+
+		- Embedding involves harmful and dangerous embeds, whether they cause brute damage, stamina damage, or a mix. This is the default behavior for embeddings, for when something is "pointy"
+
+		- Sticking occurs when an item should not cause any harm while embedding (imagine throwing a sticky ball of tape at someone, rather than a shuriken). An item is considered "sticky"
+			when it has 0 random pain chance and 0 jostling chance. It's a bit arbitrary, but fairly straightforward.
+
+		Stickables differ from embeds in the following ways:
+			-- Text descriptors use phrasing like "X is stuck to Y" rather than "X is embedded in Y"
+			-- There is no slicing sound on impact
+			-- All damage checks and bloodloss are skipped for humans
+			-- Pointy objects create sparks when embedding into a turf
+
+*/
+
+
+/datum/component/embedded
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	var/obj/item/bodypart/L
+	var/obj/item/weapon
+
+	var/embed_chance // not like we really need it once we're already stuck in but hey
+	var/fall_chance
+	var/pain_chance
+	var/pain_mult
+	var/impact_pain_mult
+	var/remove_pain_mult
+	var/rip_time
+	var/ignore_throwspeed_threshold
+	var/jostle_chance
+	var/jostle_pain_mult
+	var/pain_stam_pct
+
+	var/harmful
+	var/mutable_appearance/overlay
+
+// figure out if im gonna drop args
+/datum/component/embedded/Initialize(obj/item/I,
+			datum/thrownthing/throwingdatum,
+			embed_chance = EMBED_CHANCE,
+			fall_chance = EMBEDDED_ITEM_FALLOUT,
+			pain_chance = EMBEDDED_PAIN_CHANCE,
+			pain_mult = EMBEDDED_PAIN_MULTIPLIER,
+			remove_pain_mult = EMBEDDED_UNSAFE_REMOVAL_PAIN_MULTIPLIER,
+			//also fall pain
+			impact_pain_mult = EMBEDDED_IMPACT_PAIN_MULTIPLIER,
+			//rip_pain_mult = EMBEDDED_UNSAFE_REMOVAL_PAIN_MULTIPLIER,
+			rip_time = EMBEDDED_UNSAFE_REMOVAL_TIME,
+			ignore_throwspeed_threshold = FALSE,
+			jostle_chance = EMBEDDED_JOSTLE_CHANCE,
+			jostle_pain_mult = EMBEDDED_JOSTLE_PAIN_MULTIPLIER,
+			pain_stam_pct = EMBEDDED_PAIN_STAM_PCT)
+
+	. = ..()
+
+	if((!ishuman(parent) && !isclosedturf(parent)) || !isitem(I))
+		return COMPONENT_INCOMPATIBLE
+
+	src.embed_chance = embed_chance
+	src.fall_chance = fall_chance
+	src.pain_chance = pain_chance
+	src.pain_mult = pain_mult
+	src.remove_pain_mult = remove_pain_mult
+	src.rip_time = rip_time
+	src.ignore_throwspeed_threshold = ignore_throwspeed_threshold
+	src.jostle_chance = jostle_chance
+	src.jostle_pain_mult = jostle_pain_mult
+	src.pain_stam_pct = pain_stam_pct
+
+	src.weapon = I
+
+	if(src.pain_chance || src.jostle_chance)
+		harmful = TRUE
```

Comment:
Does this need a var? Sounds like a simple helper works

Reply:
well it's not like it'll ever change during the component's life, so one-and-done is probably better as a micro-opt